plugins/woocommerce/src/Admin/Overrides/OrderTraits.php

```
@@ -42,29 +52,43 @@ public function get_item_shipping_amount( $item ) {
-					$tax_item_id                = $tax_item->get_rate_id();
-					$tax_item_total             = isset( $tax_data['total'][ $tax_item_id ] ) ? (float) $tax_data['total'][ $tax_item_id ] : 0;
-					$total_shipping_tax_amount += $tax_item_total;
+		// Use the passed shipping_tax_amount if provided, otherwise initialize it to 0 and calculate the total shipping tax amount in the order.
+		// This is useful when calculating refunds for partial shipping tax in an order.
+		// For example, if $1 shipping tax is refunded from an order with $3 shipping tax, the remaining $2 should be distributed to the remaining items.
+		$total_shipping_tax_amount = $shipping_tax_amount ? $shipping_tax_amount : 0;
+
+		if ( null === $shipping_tax_amount ) {
+			$order_taxes         = $this->get_taxes();
+			$line_items_shipping = $this->get_items( 'shipping' );
+			foreach ( $line_items_shipping as $item_id => $shipping_item ) {
```

PHPMD (2.15.0)

Avoid unused local variables such as '$item_id'. (Unused Code Rules)

(UnusedLocalVariable)